core/src/main/scala/slinky/core/facade/Suspense.scala

```diff
@@ -16,3 +23,26 @@ object Suspense
 
   def apply(fallback: ReactElement): BuildingComponent[Nothing, js.Object] = apply(Props(fallback))
 }
+
+
+object Suspended {
+
+  class PromiseWrap[T](suspend: Future[T]) extends js.Object {
+    private val pending = suspend.toJSPromise
+    private var data: Option[T] = None
```

Comment:
this shouldn't be private as I'm finding often non-presentational logic may want to check the existence of data without triggering unnecessary exceptions to be thrown